crates/matrix-sdk/src/client/mod.rs

```diff
@@ -239,10 +239,10 @@ pub(crate) struct ClientInner {
     base_client: BaseClient,
 
     /// The Matrix versions the server supports (well-known ones only)
-    server_versions: OnceCell<Box<[MatrixVersion]>>,
+    server_versions: RwLock<Option<Box<[MatrixVersion]>>>,
```

Comment:
I think we should put those behind the same lock, as we saw in the memory store, it's tricky to remember how to avoid a deadlock when two pieces of information are accessed or updated in the same method but are living behind separate locks.

Further bellow we do acquire the locks in the same method.